modules/ting_covers/ting_covers.pages.inc

```diff
@@ -8,12 +8,19 @@
 /**
  * Return the URL for the cover of an object.
  */
-function ting_covers_objects() {
+function ting_covers_objects($http_request = TRUE, $cover_data = array()) {
   $covers = array();
   $image_styles = array();
   $missing_images_local_ids = array();
 
-  foreach ($_POST['coverData'] as $cover_info) {
+  if ($http_request) {
+    $cover_data = $_POST['coverData'];
+  }
+  elseif (!$http_request && empty($cover_data)) {
+    return $covers;
+  }
+
+  foreach ($cover_data as $cover_info) {
```

Comment:
I do not understand why these two new parameters are needed. This is a page callback and should not be called otherwise.

If we need similar functionality then some of the internals of `ting_covers_objects()` should be refactored into a separate function which can be used in the page callback and elsewhere needed.